modules/control-utility/src/test/java/org/apache/ignite/util/KillCommandsCommandShTest.java

```diff
@@ -195,4 +205,97 @@ public void testCancelUnknownContinuousQuery() {
 
         assertEquals(EXIT_CODE_OK, res);
     }
+
+    /** */
+    @Test
+    public void testCancelConsistencyMissedTask() {
+        int res = execute("--kill", "consistency");
+
+        assertEquals(EXIT_CODE_OK, res);
+    }
+
+    /** */
+    @Test
+    public void testCancelConsistencyTask() throws InterruptedException {
+        String consistencyCancheName = "consistencyCache";
+
+        CacheConfiguration<Integer, Integer> cfg = new CacheConfiguration<>();
+
+        cfg.setName(consistencyCancheName);
+        cfg.setBackups(SERVER_NODE_CNT - 1);
+
+        IgniteCache<Integer, Integer> cache = client.getOrCreateCache(cfg);
+
+        for (int i = 0; i < 10_000; i++)
+            cache.put(i, i);
+
+        CountDownLatch getLatch = new CountDownLatch(SERVER_NODE_CNT); // Each node should send a get request.
+
+        for (IgniteEx server : srvs) {
+            TestRecordingCommunicationSpi spi =
+                ((TestRecordingCommunicationSpi)server.configuration().getCommunicationSpi());
+
+            spi.blockMessages((node, message) -> {
+                if (message instanceof GridNearGetRequest) { // Get request caused by read repair operation.
+                    getLatch.countDown();
+
+                    return true; // Blocking to freeze '--consistency repair' operation.
+                }
+
+                return false;
+            });
+        }
+
+        CountDownLatch thLatch = new CountDownLatch(1);
+
+        new Thread(() -> {
+            try {
+                getLatch.await(); // Waiting for consistency repair start.
+            }
+            catch (InterruptedException e) {
+                fail();
+            }
+
+            IgnitePredicate<GridJobWorker> repairJobFilter =
+                worker -> worker.getJob() instanceof VisorConsistencyRepairTask.VisorConsistencyRepairJob;
+
+            for (IgniteEx node : srvs)
+                assertFalse(node.context().job().findJobs(repairJobFilter).isEmpty()); // Found.
+
+            int res = execute("--kill", "consistency");
+
+            assertEquals(EXIT_CODE_OK, res);
+
+            try {
+                assertTrue(GridTestUtils.waitForCondition(() -> {
+                    for (IgniteEx node : srvs)
+                        if (!node.context().job().findJobs(repairJobFilter).isEmpty())
+                            return false;
+
+                    return true;
+                }, 5000L)); // Missed.
```

Comment:
let's use `getTestTimeout()` here to deal with slow test environment.

Reply:
getTestTimeout() is a 5 minutes instead of 5 seconds I'm proposing.
And again it's a timeout need to survive on GC pause or VM freeze, we'll never wait for it at the regular run.